modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZonesUtil.java

```diff
@@ -274,16 +264,25 @@ static ByteArray zoneTopologyAugmentationVault(int zoneId) {
      * Condition for updating {@link DistributionZonesUtil#zoneScaleUpChangeTriggerKey(int)} key.
      * Update only if the revision of the event is newer than value in that trigger key.
      *
-     * @param revision Event revision.
      * @return Update condition.
      */
-    static CompoundCondition triggerKeyConditionForZonesChanges(long revision, int zoneId) {
-        return or(
-                notExists(zonesChangeTriggerKey(zoneId)),
-                value(zonesChangeTriggerKey(zoneId)).lt(ByteUtils.longToBytes(revision))
+    static CompoundCondition conditionForZoneCreation(int zoneId) {
+        return and(
+                notExists(zoneDataNodesKey(zoneId)),
+                notTombstone(zoneDataNodesKey(zoneId))
```

Comment:
Do we have test cases where ms execute invoke with this condition and zoneDataNodesKey(zoneId) is a tombstone?